src/Microsoft.AspNet.Http.Core/Extensions/MapWhenMiddleware.cs

```diff
@@ -20,27 +20,13 @@ public MapWhenMiddleware([NotNull] RequestDelegate next, [NotNull] MapWhenOption
 
         public async Task Invoke([NotNull] HttpContext context)
         {
-            if (_options.Predicate != null)
+            if (_options.Predicate(context))
```

Comment:
That's a guard on the method, not on the actual options class. Someone can still go to the options and make it null, no?